lua/entities/ent_snowball_nodamage.lua

```diff
@@ -41,19 +41,22 @@ function ENT:Draw()
 end
 
 function ENT:PhysicsCollide(data)
+	local plycol = self.Owner:GetNWVector("SnowballColor", Vector(1, 1, 1)):ToColor()
 	local pos = self.Entity:GetPos()
 	local effectdata = EffectData()
 	local fwd = self:GetOwner():GetAimVector()
 	if data.HitEntity:IsPlayer() then --push players around a small amount
 		data.HitEntity:SetVelocity(Vector(fwd * 100))
 	end
 
-	local p1 = data.HitPos + (data.HitNormal * 2)
-	local p2 = data.HitPos - (data.HitNormal * 2)
-	util.Decal("Splash.Large", p1, p2)
+	--local p1 = data.HitPos + (data.HitNormal * 2)
+	--local p2 = data.HitPos - (data.HitNormal * 2)
+	--util.Decal("Splash.Large", p1, p2)
+	util.DecalEx(Material("Splash.Large"), nil, data.HitPos, data.HitNormal, plycol, 1, 1)
```

Comment:
Should be `util.DecalMaterial` not `Material`, and also, cache it outside the function

Reply:
okay made the changes, hopefully I did them right